### PR TITLE
Feature: Easier Patient Visits

### PR DESCRIPTION
**Pull request by jniles**
This PR addresses the feature request in #3012.

**Design**
The overall design is to make patient admission and discharge much easier to do for the receptionist.  In 1.x, we previously used a separate area on the registration form, but this caused confusion and was limited to looking up patients by their name or BHIMA identifier.  Normally, returning patients provide their hospital number, so this implementation allows a patient to be discovered based on their hospital number.

We also distinguish between "Ambulatory" and "Hospitalized" patients.  The hospitalized patients will need to be discharged, whilst the ambulatory patients could automatically be discharged immediately after admission.  The rationale is that it would be difficult to know if a patient has left the hospital or not.  We may change this behavior with further testing.

This is ready for an initial review.  @mbayopanda could I get one?

### CODE REVIEW

File: client/src/modules/patients/record/units/visits.modal.html [hunk 1]
```diff
@@ -1,18 +1,43 @@
 <div class="modal-header">
   <ol class="headercrumb">
-    <li class="static"><span translate>FORM.INFO.PATIENTS</span></li>
```
Comment (mbayopanda) [inline client/src/modules/patients/record/units/visits.modal.html:3]: Nice catch 👍

File: client/src/modules/patients/record/units/visits.service.js [hunk 2]
```diff
@@ -67,21 +70,19 @@ function VisitService($http, util, Modal) {
...
-      animation : false,
       resolve : {
-        patient : function patientProvider() { return patientUuid; },
-        isAdmission : function isAdmissionProvider() { return isAdmission; },
-        currentVisit : function currentVisitProvider() { return currentVisit; }
-      }
+        patient : () => patientUuid,
+        isAdmission : () => isAdmission,
+        currentVisit : () => currentVisit,
+      },
```
Comment (mbayopanda) [inline client/src/modules/patients/record/units/visits.service.js:86]: 👍 much better

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

I have noticed just two things in the ui : 
- 1. After to have define a visit, the modal is not automatically updated with information about the visit, we have to refresh the page in the browser for seeing that the patient is now "Admit in the hospital (En cours)".

- 2. The "Annuler" button could be "Fermer", this last one is more meaningful according the context of the modal.

patient admission

**Comment by jniles** [comment]
@mbayopanda, thank you for the review.

It seems that our clients have multiple different thoughts about how to model diagnoses and admissions in BHIMA.  I'll wait for the discussion to finish before too many more changes are made here.

**Comment by jniles** [comment]
So, I've heard back from our clients and they request the following information be included in a visit:


Date de   consultation | Date of consultation  consultation
-- | --
N° Registre de consultation Médecins | No. Register of consultation by Doctors
N° entrant    Hospitalisation | Incoming number  Hospitalization
N° BHIMA | BHIMA NUMBER
Nom-Post nom | Name-Post name name
Prénom | First name
Sexe | Gender
Age : années | Age: years
Rue/Village | Street/Village
Quartier/ Secteur | Neighbourhood/ Sector
Z | Vanga Zone
HZ | Other Zones
Référé CS de provenance | Referred health center      of origin
Diagnostic d'entrée et Autre diagnostic: | Admission diagnosis and Other diagnosis:
Treatment: | Treatment:
Date      d'entrée | Date of admit
Date de      Sortie | Date of discharge     Exit
Nombre de Jours | Number Hospitalized Days
Etat a la sortie (G/T/D/E) | Discharge status      (G/T/D/E)
Guéri | Healed
Transféré | Transferred
Mort | Death
S'est évadé sans décharge officielle | Evaded without official discharge
Pavillion | Hospital Wing/Pavillion
N° Lit | Bed number
Plaintes | Complaints
Laboratoire | Lab
Diagnostic   : | Diagnosis:
Traitement : | Treatment:



As well as:

Pavillion | Hospital Wing/Pavillion
-- | --
N° Lit | Bed number
Plaintes | Complaints
Laboratoire | Lab
Diagnostic   : | Diagnosis:
Traitement : | Treatment:

**Comment by mbayopanda** [comment]
@jniles what is the status of this PR ?

**Comment by jniles** [comment]
Our Vanga clients are still wanting this PR to land, but with the added features.  Let me make it into a feature branch, like Cost Centers and close the PR.

**Comment by jniles** [comment]
Alright, changed the base branch.  Once the conflicts are fixed, we can merge this.

**Comment by jniles** [comment]
@mbayopanda changes made!  Can be merged into the feature branch anytime.

**Pull request merged**